### ISSUE TITLE
Ionic + AWS Amplify breaks on production build

### ISSUE THREAD

**Issue by jonjet** [aws-amplify/amplify-js#2275]
**Describe the bug**
Compiled production build for Ionic with AWS Amplify module ends up with the app stuck on the splash screen. Troubleshooting the issue resulted in the following error being displayed:-

Uncaught Error: Cannot use e "__Schema" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results.
    at t.a (vendor.js:1)
    at f (vendor.js:1)
    at c (vendor.js:1)
    at B (vendor.js:1)
    at q (vendor.js:1)
    at new a (vendor.js:1)
    at a (vendor.js:1)
    at Object.<anonymous> (vendor.js:1)
    at t (vendor.js:1)
    at Object.<anonymous> (vendor.js:1)

**To Reproduce**
1. App on Ionic 3 with the following dependencies: "aws-amplify": "^1.1.12", "aws-amplify-angular": "^2.0.15",
2. Build and run application with ionic cordova run android --prod --device
3. App loads and gets stuck on splash screen
4. Review Chrome Inspector and error is displayed

**Expected behavior**
Failed to execute production build of Ionic 3 application when AWS Amplify dependency is used.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Redmi 4X
 - OS: Android 7.1.2, API 25

**Additional context**
package.json
  "scripts": {
    "start": "ionic-app-scripts serve",
    "clean": "ionic-app-scripts clean",
    "build": "ionic-app-scripts build",
    "lint": "ionic-app-scripts lint"
  },
  "resolutions": {
    "graphql": "^0.13.0"
  },
  "dependencies": {
    "@angular/animations": "5.2.11",
    "@angular/common": "5.2.11",
    "@angular/compiler": "5.2.11",
    "@angular/compiler-cli": "5.2.11",
    "@angular/core": "5.2.11",
    "@angular/forms": "5.2.11",
    "@angular/http": "5.2.11",
    "@angular/platform-browser": "5.2.11",
    "@angular/platform-browser-dynamic": "5.2.11",
    "@ionic-native/barcode-scanner": "^4.18.0",
    "@ionic-native/core": "~4.15.0",
    "@ionic-native/screen-orientation": "^4.18.0",
    "@ionic-native/splash-screen": "~4.15.0",
    "@ionic-native/status-bar": "~4.15.0",
    "@ionic/pro": "2.0.3",
    "@ionic/storage": "2.2.0",
    "aws-amplify": "^1.1.12",
    "aws-amplify-angular": "^2.0.15",
    "cordova-android": "7.1.4",
    "cordova-ios": "4.5.5",
    "cordova-plugin-add-swift-support": "1.7.2",
    "cordova-plugin-device": "2.0.2",
    "cordova-plugin-ionic-keyboard": "2.1.3",
    "cordova-plugin-ionic-webview": "2.2.5",
    "cordova-plugin-screen-orientation": "3.0.1",
    "cordova-plugin-splashscreen": "5.0.2",
    "cordova-plugin-statusbar": "2.4.2",
    "cordova-plugin-whitelist": "1.3.3",
    "cryptocurrency-icons": "^0.9.3",
    "es6-promise-plugin": "4.2.2",
    "ionic-angular": "3.9.2",
    "ionic-marquee": "^0.3.0",
    "ionicons": "3.0.0",
    "phonegap-plugin-barcodescanner": "8.0.1",
    "rxjs": "5.5.11",
    "sw-toolbox": "3.6.0",
    "zone.js": "0.8.26"
  },
  "devDependencies": {
    "@ionic/app-scripts": "3.2.0",
    "@types/node": "^10.12.12",
    "typescript": "~2.6.2"
  },
  "description": "The easiest way to get cryptocurrency to your wallet!",
  "cordova": {
    "plugins": {
      "cordova-plugin-add-swift-support": {},
      "cordova-plugin-device": {},
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-ionic-webview": {
        "ANDROID_SUPPORT_ANNOTATIONS_VERSION": "27.+"
      },
      "cordova-plugin-screen-orientation": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-whitelist": {},
      "es6-promise-plugin": {},
      "phonegap-plugin-barcodescanner": {
        "ANDROID_SUPPORT_V4_VERSION": "27.+"
      }
    },
    "platforms": [
      "ios",
      "android"
    ]
  }
}

***You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app.***

**Comment by haverchuck** [comment]
@jonjet  - I was about to reproduce this issue, and found numerous discussions of it issue on various GraphQL related forums/threads.  

So I tried one of the recommended solutions...
```
$ NODE_ENV=production
$ ionic cordova run android --prod
```
... and this seemed to work (at least on the emulator).

Any chance you could give this a try?

**Comment by jonjet** [comment]
Thanks @haverchuck for checking this out. I've tried out your recommended solutions and I am still getting the same error.

The only way I manage to get it working is using the following solutions in another thread:-
`$ ionic cordova run android --aot --uglifyjs --minifycss --optimizeJs --device`

When I run the with the --minifyjs flag, it reproduces the same error as --prod flag

**Comment by haverchuck** [comment]
@jonjet  - I tested with an actual Android device, and with the package.json you posted above.

I was able to reproduce the issue with the following commands:

- `ionic cordova run android --prod --device`
- `ionic cordova run android --aot --uglifyjs --minifyjs  --minifycss --optimizeJs --device`

I was able to get the sample app to build correctly with the following variations:

- `NODE_ENV=production ionic cordova run android --prod --device`
- `NODE_ENV=production ionic cordova run android --aot --uglifyjs --minifyjs  --minifycss --optimizeJs --device`

I know you said above that setting the NODE_ENV didn't help you, but I'm not sure what else it could be.  I tried the two commands you cited as failing, and prepending the NODE_ENV=production statement fixed both of them.  

Maybe something is happening with environmental config files or the like?

**Comment by jonjet** [comment]
@haverchuck thanks for following up on this issue and providing a solution on this!

I have revisited the application built in Ionic 3 and managed to get the builds to work properly on iOS and Android with the variations of NODE_ENV=production provided. Both iOS/Android builds compiled with no errors, and loads perfectly fine after launch. 

I did not have any resolutions defined for GraphQL. NPM package versions used below for reference.

>     "aws-amplify": "^1.1.18",
>     "aws-amplify-angular": "^2.1.5",

Android Build
`NODE_ENV=production ionic cordova run android --prod --device`

iOS Build
`NODE_ENV=production ionic cordova run ios --prod --device`
